src/parser/grammar.y

```diff
@@ -277,13 +278,33 @@ edge(A) ::= LEFT_BRACKET UQSTRING(B) properties(C) RIGHT_BRACKET . {
 }
 
 // Edge with label [:label]
-edge(A) ::= LEFT_BRACKET COLON UQSTRING(B) edgeLength(C) properties(D) RIGHT_BRACKET . { 
-	A = New_AST_LinkEntity(NULL, B.strval, D, N_DIR_UNKNOWN, C);
+edge(A) ::= LEFT_BRACKET edgeLabels(B) edgeLength(C) properties(D) RIGHT_BRACKET . { 
+	A = New_AST_LinkEntity(NULL, B, D, N_DIR_UNKNOWN, C);
 }
 
 // Edge with alias and label [alias:label]
-edge(A) ::= LEFT_BRACKET UQSTRING(B) COLON UQSTRING(C) properties(D) RIGHT_BRACKET . { 
-	A = New_AST_LinkEntity(B.strval, C.strval, D, N_DIR_UNKNOWN, NULL);
+edge(A) ::= LEFT_BRACKET UQSTRING(B) edgeLabels(C) properties(D) RIGHT_BRACKET . { 
+	A = New_AST_LinkEntity(B.strval, C, D, N_DIR_UNKNOWN, NULL);
+}
+
+
+%type edgeLabel {char*}
+// Single label
+edgeLabel(A) ::= COLON UQSTRING(B) . {
+	A = B.strval;
+}
+
+%type edgeLabels {char**}
+edgeLabels(A) ::= edgeLabel(B) . {
+	A = array_new(char*, 1);
+	A = array_append(A, B);
+}
+
+// Multiple labels
+edgeLabels(A) ::= edgeLabels(B) PIPE edgeLabel(C) . {
```

Comment:
For multiple types, the Cypher standard specifies one colon for the multi-relationship format:
`(a)-[r:TYPE1|TYPE2]->(b)`
Though in practice Neo4j also accepts multiple colons like here.

Reply:
I had no idea, it seems logical to me that the language should allow you to specify as many relationship types as you want

Reply:
Ah, sorry to be unclear! You can specify as many relationship types as you want, but the colon is only used once:
`r:TYPE1|TYPE2|TYPE3]`
is fine.